copylot/hardware/prime_bsi_camera/camera.py

```diff
@@ -0,0 +1,118 @@
+from pyvcam import pvc, constants
+from pyvcam.camera import Camera
+
+
+class PrimeBSICamera:
+    def __init__(
+        self,
+        scan_mode,
+        scan_dir,
+        binning: tuple,
+        trigger_mode,
+        exposure_out,
+        enable_metadata: bool = True,
+    ):
+        """
+
+        Parameters
+        ----------
+        scan_mode
+        scan_dir
+        binning : tuple
+            (x_bin, y_bin). If you want to set square
+            binning, you should pass equal values for
+            x_bin and y_bin.
+        enable_metadata : bool
+            Toggles camera metadata on and off. By default
+            it is True.
+        """
+        pvc.init_pvcam()
+
+        self.cam.prog_scan_mode = scan_mode
+        self.cam.prog_scan_dir = scan_dir
+        self.cam.binning = binning
+        self.cam.exp_mode = trigger_mode
+        self.cam.exp_out_mode = exposure_out
+
+        self.cam.set_param(constants.PARAM_METADATA_ENABLED, enable_metadata)
```

Comment:
Hey @AhmetCanSolak I'll take a more detailed look at this code today. Quick question - it looks to me that here `self.cam` is used before it is assigned. I think you're missing something like 
```
self.cam = next(Camera.detect_camera())
```
is that right?

Reply:
Yes that part is still incomplete. I want to implement it in a way to be able to choose the specific camera when you are initiating this class but haven't got around it yet.

Reply:
Got it, thanks. I also think that's useful, I'll try to implement that.